services/cd-service/pkg/repository/transformer.go

```diff
@@ -839,13 +839,22 @@ func (c *ReleaseTrain) Transform(ctx context.Context, state *State) (string, err
 	if envConfig.Upstream == nil {
 		return fmt.Sprintf("Environment %q does not have upstream configured - exiting.", targetEnvName), nil
 	}
+	var upstreamLatest = envConfig.Upstream.Latest
 	var upstreamEnvName = envConfig.Upstream.Environment
-	if upstreamEnvName == "" {
-		return fmt.Sprintf("Environment %q does not have upstream environment configured - exiting.", targetEnvName), nil
+
+	if !upstreamLatest && upstreamEnvName == "" {
```

Comment:
please check the reverse too:
It should not be allowed to have both latest and environment set at the same time.

Reply:
good idea :+1: